api/webapp/clientapi/dom/DataRegion.js

```diff
@@ -3161,9 +3161,9 @@ if (!LABKEY.DataRegions) {
                             $.each(skipPrefixSet, function (j, skipPrefix) {
                                 if (LABKEY.Utils.isString(skipPrefix)) {
 
-                                    // Special prefix that should remove all filters, but no other parameters
+                                    // Special prefix that should remove all filters, but no other parameters for the current grid
                                     if (skipPrefix.indexOf(ALL_FILTERS_SKIP_PREFIX) === (skipPrefix.length - 2)) {
-                                        if (key.indexOf('~') > 0) {
+                                        if (key.indexOf(region.name) >= 0 && key.indexOf('~') > 0) {

```

Comment:
This looks like a good approach, but if we had `query` and `query2` on the same page, it would still get confused, or `mylist` and `list`

```suggestion
                                        if (key.indexOf(region.name + '.') == 0 && key.indexOf('~') > 0) {

```

Reply:
I have multiple lists and queries on the page with titles 'query' or 'list' and it does not get confused since region.name is unique within a given page. For example, for the same list displayed twice on the page with title 'My List', region.name is list4 and list5, and for Query wp, it doesn't use title either, region.name is qwp7 or qwp8, for example.

Reply:
Yes, but I bet that if you manually set `[dataRegionName](https://www.labkey.org/download/clientapi_docs/javascript-api/symbols/LABKEY.QueryWebPart.html#constructor)` on two different QueryWebParts on the same page to `list` and `mylist`, for example, they'd get confused.

Reply:
Got it! Thanks for clarifying